lib/stores/FileStore.js

```diff
@@ -45,45 +48,25 @@ class FileStore extends DataStore {
     /**
      * Create an empty file.
      *
-     * @param  {object} req http.incomingMessage
      * @param  {File} file
      * @return {Promise}
      */
-    create(req) {
+    create(file) {
         return new Promise((resolve, reject) => {
-            const upload_length = req.headers['upload-length'];
-            const upload_defer_length = req.headers['upload-defer-length'];
-            const upload_metadata = req.headers['upload-metadata'];
-
-            if (upload_length === undefined && upload_defer_length === undefined) {
-                return reject(ERRORS.INVALID_LENGTH);
-            }
-
-            let file_id;
-            try {
-                file_id = this.generateFileName(req);
-            }
-            catch (generateError) {
-                log('[FileStore] create: check your namingFunction. Error', generateError);
-                return reject(ERRORS.FILE_WRITE_ERROR);
-            }
-
-            const file = new File(file_id, upload_length, upload_defer_length, upload_metadata);
-            return fs.open(`${this.directory}/${file.id}`, 'w', (err, fd) => {
+            return fs.open(path.join(this.directory, file.id), 'w', async(err, fd) => {
```

Comment:
Rewriting this to use the fs Promise API from Node was not a goal of this PR, I assume?

Reply:
One 1.0 hits we should put official supported node versions clearly in the readme.